runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/HotKeyLogger.java

```diff
@@ -47,10 +46,31 @@
 
   /** Logs a detection of the hot key every 5 minutes. */
   public void logHotKeyDetection(String userStepName, Duration hotKeyAge) {
+    logHotKeyDetection(userStepName, hotKeyAge, null);
```

Comment:
Note that `null` is a valid key type and using null to differentiate between log without key and log with key will miss this usecase which is why I proposed the two suggestions below:
```suggestion
    if (isThrottled()) {
      return;
    }
    LOG.warn(
        "A hot key was detected in step '{}' with age of '{}'. This is "
            + "a symptom of key distribution being skewed. To fix, please inspect your data and "
            + "pipeline to ensure that elements are evenly distributed across your key space.",
        userStepName,
        TimeUtil.toCloudDuration(hotKeyAge));
```